chunk.go

```diff
@@ -0,0 +1,165 @@
+package recordio
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/binary"
+	"fmt"
+	"hash/crc32"
+	"io"
+
+	"github.com/golang/snappy"
+)
+
+// A Chunk contains the Header and optionally compressed records.
+type Chunk struct {
+	records [][]byte
+	size    int // sum of record lengths.
+}
+
+func newChunk() *Chunk {
```

Comment:
Ok for me either way, just a suggestion: Maybe we don't need this function, since there is no argument to the function. We can just use `&Chunk{}` where needed.